src/avro/src/reader.rs

```diff
@@ -712,6 +712,15 @@ impl<'a> SchemaResolver<'a> {
                     {
                         SchemaPieceOrNamed::Piece(SchemaPiece::Bytes)
                     }
+                    // TODO [btv] We probably want to rethink what we're doing here, rather than just add
+                    // a new branch for every possible "logical" type. Perhaps logical types with the
+                    // same underlying type should always be resolvable to the reader schema's type?
+                    (SchemaPiece::Json, SchemaPiece::Json) => {
+                        SchemaPieceOrNamed::Piece(SchemaPiece::Json)
+                    }
+                    (SchemaPiece::Uuid, SchemaPiece::Uuid) => {
+                        SchemaPieceOrNamed::Piece(SchemaPiece::Uuid)
+                    }
```

Comment:
could we just do `(left, right) if left == right => SchemaPieceOrNamed::Piece(left),`?

Reply:
Yeah... probably? I'm actually not sure why we're not already doing that.

Reply:
Oh, I believe it's not actually valid, because equality on schema pieces doesn't quite do what you would expect.

For example, two arrays that reference the named schema fragment at index 3 would compare equal, even if that actually references a different schema in the two contexts.

Reply:
oh, that's too bad.